operations/mimir-mixin/dashboards/compactor.libsonnet

```diff
@@ -40,6 +162,106 @@ local utils = import 'mixin-utils/utils.libsonnet';
+      .addPanel(
+        $.panel('Longest time since last successful run') +
+        $.panelDescription(
+          'Longest time since last successful run',
+          |||
+            Displays the amount of time since the most recent successful execution
+            of the compactor.
+            The value shown will be for the compactor replica that has the longest time since its
+            last successful run.
+            The table to the right shows a summary for all compactor replicas.
+
+            If there is no time value, one of the following messages might appear:
+
+            - If you see "No compactor data" in this panel, that means that no compactors are active yet.
+
+            - If you see "No successful runs" in this panel, that means that compactors are active, but none 
+              of them were successfully executed yet.
+
+            These might be expected - for example, if you just recently restarted your compactors, they might not 
+            yet be broadcasting metrics, or they might not have had a chance to complete their first compaction run. 
+            However, if these messages persist, you should check the health of your compactors.
+          |||
+        ) +
+        $.newStatPanel(lastRunQuery, unit='s') + {
+          options: {
+            reduceOptions: {
+              values: false,
+              calcs: ['first'],
+              fields: '/^Last run$/',
+            },
+            textMode: 'value',
+          },
+          targets: [target { format: 'table', instant: true } for target in super.targets],
+          transformations:
+            //[transformation('merge')] +
+            lastRunCommonTransformations,
+          fieldConfig: super.fieldConfig + {
+            defaults: super.defaults {
+              noValue: 'No compactor data',
+            },
+            overrides: [
+              overrideFieldByName('Last run', [
+                overrideProperty('custom.width', 74),
+                overrideProperty('mappings', [
+                  mappingRange('-Infinity', 0, { color: 'text', text: 'No successful runs' }),
+                ]),
+                overrideProperty('color', { mode: 'thresholds' }),
+                overrideProperty('thresholds', { mode: 'absolute', steps: lastRunThresholds.steps }),
+              ]),
+            ],
+          },
+        },
+      )
+      .addPanel(
+        $.panel('Last successful run per-compactor replica') +
+        $.panelDescription(
+          'Last successful run per-compactor replica',
+          |||
+            Displays the compactor replicas, and for each, shows how long it has been since
+            its last successful compaction run.
+
+            The value in the status column is based on how long it has been since the last successful compaction.
+
+            - Okay: less than 2 hours
+            - Delayed: more than 2 hours
+            - Late: more than 6 hours
+            - Very late: more than 12 hours
```

Comment:
Since the actual logic is based on variables, I would disable number of hours here based on the actual variable values.

Reply:
:smiley: